packages/shared/routes/dashboard/wallet/views/Send.svelte

```diff
@@ -98,27 +120,150 @@
+        if(remainderData?.address)
+            return { remainderAddress: remainderData?.address }
+
+        const txData = eventData as PreparedTransactionEvent
+        if(!(txData?.inputs && txData?.outputs) || (txData?.inputs.length <= 0 || txData?.outputs.length <= 0))
+            return {}
+
+        const numOutputs = txData.outputs.length
+        if(numOutputs === 1) {
+            return {
+                toAddress: txData.outputs[0].address,
+                toAmount: txData.outputs[0].amount
+            }
+        } else if(numOutputs > 1) {
+            return {
+                toAddress: txData.outputs[0].address,
+                toAmount: txData.outputs[0].amount,
+
+                remainderAddress: txData.outputs[numOutputs - 1].address,
+                remainderAmount: txData.outputs[numOutputs - 1].amount
+            }
+        } else {
+            return txData
+        }
+    }
+
+    const handleTransferState = (state: TransferState): void => {
+        if(!state)
+            return
+
+        const _onCancel = () => {
+            isTransferring.set(false)
+            transferState.set(null)
+
+            clearSendParams(selectedSendType === SEND_TYPE.INTERNAL)
             closePopup()
+
+            if(get(displayNotifications).length === 0)
+                showAppNotification({
+                    type: 'error',
+                    message: locale('error.send.transaction'),
+                })
+        }
+
+        const { data, type } = state
+        switch(type) {
+            default:
+                if(ledgerAwaitingConfirmation) {
+                    ledgerAwaitingConfirmation = false
+
+                    closePopup()
+                }
+
+                break
+
+            case TransferProgressEventType.GeneratingRemainderDepositAddress:
+                transactionEventData = data
+
+                /**
+                 * NOTE: The break statement is omitted in this case to allow the next block of code
+                 * (under SigningTransaction) to be executed.
+                 */
+
+            case TransferProgressEventType.SigningTransaction:
+                ledgerAwaitingConfirmation = true
+
+                openPopup({
+                    type: 'ledgerTransaction',
+                    hideClose: true,
+                    props: {
+                        onCancel: _onCancel,
+                        ...handleTransactionEventData(transactionEventData)
+                    }
+                })
+
+                break
+
+            case TransferProgressEventType.PreparedTransaction:
+                /**
+                 * CAUTION: The Ledger confirmation doesn't always trigger
+                 * the popup to close, so it is programmatically enforced here.
+                 */
+                if(get(popupState).active)
+                    closePopup()
+
+                transactionEventData = data
+
+                break
         }
     }
+
+    $: if(!$isSoftwareProfile)
```

Comment:
we should use `get(isSoftwareProfile)` here to avoid making the the statement reactive